smqtk_detection/impls/detect_image_objects/resnet_frcnn.py

```diff
@@ -0,0 +1,201 @@
+import numpy as np
+from typing import Tuple, Iterable, Dict, Hashable, List
+import importlib.util
+
+try:
+    import torch  # type: ignore
+    import torchvision.models as models  # type: ignore
+    from torchvision import transforms  # type: ignore
+    from torchvision.ops import boxes as box_ops  # type: ignore
+    import torch.nn.functional as F  # type: ignore
+    from torchvision.models.detection.roi_heads import RoIHeads  # type: ignore
+except ModuleNotFoundError:
+    pass
+
+from smqtk_detection.interfaces.detect_image_objects import DetectImageObjects
+from smqtk_detection.utils.bbox import AxisAlignedBoundingBox
+
+import warnings
+warnings.filterwarnings("ignore", category=UserWarning)
+
+
+class ResNetFRCNN(DetectImageObjects):
+    """
+    ``DetectImageObjects`` implementation using ``torchvision``'s Faster R-CNN
+    with a ResNet-50-FPN backbone, pretrained on COCO train2017.
+    """
+
+    def __init__(
+        self,
+        box_thresh: float = 0.05,
+        num_dets: int = 100,
+        img_batch_size: int = 1,
+        use_cuda: bool = True,
+    ):
+        """
+        :param box_thresh: Confidence threshold for detections.
+        :param num_dets: Maximum number of detections per image.
+        :param img_batch_size: Batch size in images for inferences.
+        :param use_cuda: Attempt to use a cuda device for inferences. If no
+            device is found, CPU is used.
+        """
+
+        self.box_thresh = box_thresh
+        self.num_dets = num_dets
+        self.img_batch_size = img_batch_size
+
+        self.model = models.detection.fasterrcnn_resnet50_fpn(
+            pretrained=True,
+            progress=False,
+            box_detections_per_img=num_dets,
+            box_score_thresh=box_thresh
+        )
+
+        self.model = self.model.eval()
+
+        if torch.cuda.is_available() and use_cuda:
+            self.use_cuda = True
+            self.model = self.model.cuda()
+        else:
+            self.use_cuda = False
+
+        self.model_loader = transforms.Compose([
+            transforms.ToPILImage(),
+            transforms.ToTensor(),
+        ])
+
+        RoIHeads.postprocess_detections = _postprocess_detections
```

Comment:
Is there a known way to do this but only for this model or network instance? I'm assuming this impacts all detectors that might be run in the same process context.